pkg/debug/checkups/processes.go

```diff
@@ -0,0 +1,70 @@
+package checkups
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+type Processes struct {
+	summary     string
+	kolideCount int
+	data        []string // FIXME: this should be more structured
+}
+
+func (c *Processes) Name() string {
+	return "Process Report"
+}
+
+func (c *Processes) Run(ctx context.Context, fullWriter io.Writer) error {
+	if fullWriter == nil {
+		return errors.New("missing io.Writer")
+	}
+
+	c.data = []string{}
+
+	ps, err := process.ProcessesWithContext(ctx)
+	if err != nil {
+		return fmt.Errorf("getting process list: %w", err)
+	}
+
+	for _, p := range ps {
+		exe, _ := p.Exe()
+
+		// full gets the full process logs. This lets us examine them for processes that might interfere.
+		fmt.Fprintf(fullWriter, p.String())
+
+		if strings.Contains(strings.ToLower(exe), "kolide") {
+			c.kolideCount += 1
+		}
+
+		// FIXME: this should be more structured
+		c.data = append(c.data, p.String())
+	}
+
+	return nil
+}
+
+func (c *Processes) ExtraFileName() string {
+	return "process.txt"
+}
+
+func (c *Processes) Status() Status {
+	if c.kolideCount > 2 {
```

Comment:
Don't we expect 3 for macOS, and 2 for Windows/Linux?

Reply:
Dunno, I didn't want to think super hard. This felt okay as a high level pass/fail

Reply:
Sorry, I didn't write my comment very clearly -- I meant that I think this will fail on Windows/Linux when it shouldn't, because we only have have the extra (3rd) sudo process on macOS.

Reply:
Hrm. We should have launcher, desktop, and osquery.

Though in some cases we won't have desktop.

I updated to `>= 2` because it's mostly meant as a very coarse check.

Reply:
Oh, the thing that confused me was the check -- `if strings.Contains(strings.ToLower(exe), "kolide") {` -- I read too quickly and didn't pick up on that being in the directory part of the filepath. This makes sense to me now. 👍